pipeline/DESIGN.md

```diff
@@ -0,0 +1,212 @@
+| stream_cid  | bytes             | Cid of the stream                                                                               |
+| stream_type | u8                | Type of the stream, see [stream type values](#stream-types)                                     |
+| controller  | string            | Controller of the stream                                                                        |
+| dimensions  | map(string,bytes) | Set of key values dimension pairs of the stream                                                 |
+| event_cid   | bytes             | Cid of the event                                                                                |
+| event_type  | u8                | Type of the event, see [event type values](#event-types)                                        |
+| state       | bytes             | JSON encoding of the event state                                                                |
+
+### Transformation
+
+Computes the singular head that is the canonical state of the stream.
+
+
+## Enumeration Values
+
+### Stream Types
+
+| name                    | code | description                                                | specification                                                                                       |
+| ----                    | ---- | -----------                                                | -------------                                                                                       |
+| Tile                    | 0x00 | A stream type representing a json document                 | https://cips.ceramic.network/CIPs/cip-8                                                             |
+| CAIP-10 Link            | 0x01 | Link blockchain accounts to DIDs                           | https://cips.ceramic.network/CIPs/cip-7                                                             |
+| Model                   | 0x02 | Defines a schema shared by group of documents in ComposeDB | https://github.com/ceramicnetwork/js-ceramic/tree/main/packages/stream-model                        |
+| Model Instance Document | 0x03 | Represents a json document in ComposeDB                    | https://github.com/ceramicnetwork/js-ceramic/tree/main/packages/stream-model-instance               |
+| UNLOADABLE              | 0x04 | A stream that is not meant to be loaded                    | https://github.com/ceramicnetwork/js-ceramic/blob/main/packages/stream-model/src/model.ts#L163-L165 |
+| EventId                 | 0x05 | An event id encoded as a cip-124 EventID                   | https://cips.ceramic.network/CIPs/cip-124                                                           |
+
+Source https://cips.ceramic.network/CIPs/cip-59#streamid-multicodec
+
+### Event Types
+
+| name | code | description                                      |
+| ---- | ---- | -----------                                      |
+| Data | 0x00 | An event containing data for the stream          |
```

Comment:
what about init?

Reply:
Yes, its only for conclusion events and its derivatives. Raw events do not have an event type enum. Its not clear yet how much we should explode the raw event table to have fields about the event itself. For now it has only the cids and the car file.